algorithms/linfa-ensemble/src/ensemble.rs

```diff
@@ -85,39 +87,45 @@ where
     }
 }
 
-pub struct EnsembleLearnerParams<P> {
+pub struct EnsembleLearnerParams<P, R: Rng + Clone> {
     pub ensemble_size: usize,
     pub bootstrap_proportion: f64,
-    pub model_params: Option<P>,
+    pub model_params: P,
+    pub rng: R
+}
+
+impl<P> EnsembleLearnerParams<P, ThreadRng> {
+    pub fn new(model_params: P) -> EnsembleLearnerParams<P, ThreadRng> {
+        return Self::new_fixed_rng(model_params, rand::thread_rng())
+    }
 }
 
-impl<P> EnsembleLearnerParams<P> {
-    pub fn new() -> EnsembleLearnerParams<P> {
+impl<P, R: Rng + Clone> EnsembleLearnerParams<P, R> {
+    pub fn new_fixed_rng(model_params: P, rng: R) -> EnsembleLearnerParams<P, R> {
         EnsembleLearnerParams {
             ensemble_size: 1,
             bootstrap_proportion: 1.0,
-            model_params: None,
+            model_params: model_params,
+            rng: rng
         }
     }
 
-    pub fn ensemble_size(&mut self, size: usize) -> &mut EnsembleLearnerParams<P> {
+    pub fn ensemble_size(&mut self, size: usize) -> &mut EnsembleLearnerParams<P, R> {
+        assert!(size > 0, "ensemble_size cannot be less than 1. Ensembles must consist of at least one model.");
```

Comment:
In `linfa` we generally use the consuming builder pattern rather than the `&mut` builder pattern. Also, instead of using `assert` to check param values can you use the `ParamGuard` pattern? You can use [the DBSCAN params](https://github.com/rust-ml/linfa/blob/master/algorithms/linfa-clustering/src/dbscan/hyperparams.rs#L63) as an example. The gist is that you use the `ParamGuard` trait to convert your current param type to a validated and immutable newtype of the same params, and then implement `Fit` on the newtype.  This allows param validation to be done without panicking.